cmd/argoexec/commands/data.go

```diff
@@ -3,18 +3,37 @@ package commands
 import (
 	"context"
 
+	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
 func NewDataCommand() *cobra.Command {
 	command := cobra.Command{
 		Use:   "data",
 		Short: "Process data",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := context.Background()
-			wfExecutor := initExecutor()
-			return wfExecutor.Data(ctx)
+		Run: func(cmd *cobra.Command, args []string) {
+			ctx := cmd.Context()
+			err := execData(ctx)
+			if err != nil {
+				log.Fatalf("%+v", err)
+			}
 		},
 	}
 	return &command
 }
+
+func execData(ctx context.Context) error {
+	wfExecutor := initExecutor()
+	defer wfExecutor.HandleError(ctx)
+
+	// Create a new empty (placeholder) task result with LabelKeyReportOutputsCompleted set to false.
+	wfExecutor.InitializeOutput(ctx)
+	defer wfExecutor.FinalizeOutput(ctx) //Ensures the LabelKeyReportOutputsCompleted is set to true.
```

Comment:
One thing I'm looking at is the `ctx` in both this and resource.go. If you look at wait.go, there is [this](https://github.com/argoproj/argo-workflows/blob/main/cmd/argoexec/commands/wait.go#L30) comment about not allowing the standard `ctx` object to prevent capture of results, parameters, artifacts, or defers. That is why that code uses `context.Background()`. I believe if we use a context that gets cancelled, then the defer to `FinalizeOutput()` will not do what it needs to do.

Reply:
Yeah, I think we need `context.Background()`

Reply:
Yes, I agree with you! Thanks